stereo_image_proc/src/nodelets/point_cloud2.cpp

```diff
@@ -174,48 +175,34 @@ void PointCloud2Nodelet::imageCb(const ImageConstPtr& l_image_msg,
   points_msg->header = disp_msg->header;
   points_msg->height = mat.rows;
   points_msg->width  = mat.cols;
-  points_msg->fields.resize (4);
-  points_msg->fields[0].name = "x";
-  points_msg->fields[0].offset = 0;
-  points_msg->fields[0].count = 1;
-  points_msg->fields[0].datatype = PointField::FLOAT32;
-  points_msg->fields[1].name = "y";
-  points_msg->fields[1].offset = 4;
-  points_msg->fields[1].count = 1;
-  points_msg->fields[1].datatype = PointField::FLOAT32;
-  points_msg->fields[2].name = "z";
-  points_msg->fields[2].offset = 8;
-  points_msg->fields[2].count = 1;
-  points_msg->fields[2].datatype = PointField::FLOAT32;
-  points_msg->fields[3].name = "rgb";
-  points_msg->fields[3].offset = 12;
-  points_msg->fields[3].count = 1;
-  points_msg->fields[3].datatype = PointField::FLOAT32;
-  //points_msg->is_bigendian = false; ???
-  static const int STEP = 16;
-  points_msg->point_step = STEP;
-  points_msg->row_step = points_msg->point_step * points_msg->width;
-  points_msg->data.resize (points_msg->row_step * points_msg->height);
```

Comment:
In `PointCloud2Modifier::setPointCloud2FieldsByString(int n_fields, ...)`,
`data.resize()` is called at last column.
https://github.com/ros/common_msgs/blob/indigo-devel/sensor_msgs/include/sensor_msgs/impl/point_cloud2_iterator.h#L227
So I think it doesn't need to call `data.resize()` expressly.